### ISSUE TITLE
Proposal: Rename the project "aqua"

### ISSUE THREAD

**Issue by suzuki-shunsuke** [aquaproj/aqua#881]
## Feature Overview

Rename the project `aqua`.

## Why is the change needed?

> Please explain the problem you want to solve.

The googleability of `aqua` is low, so it is difficult to search the information about aqua using Google or another similar Internet search engine.

## Does the change include Breaking Changes?

Yes. You have to fix codes.

* aqua
* aqua-installer
* aqua-renovate-config

---

* aqua.yaml
* GitHub Actions
* renovate.json
* install scripts

**Comment by suzuki-shunsuke** [comment]
I have no idea of new name yet.

**Comment by 4513ECHO** [comment]
How about `'turquoise'` ( synonym of aqua ) or `'suiteki'` ( aqua in Japanese ) ?
I may not good at be naming, but these is some ways of thinking.

**Comment by suzuki-shunsuke** [comment]
Thank you for your proposal.

**Comment by suzuki-shunsuke** [comment]
I would like to keep names of GitHub Organization, Repository, and command.
So how about combining any word and `Aqua`?

---

* Homebrew: brew
* cue-lang: cue

---

* AquaVM
* AquaCLIVM
* AquaSky
* ~AquaStar: This name is used https://aqua-star.co.jp/~
* AquaLite: https://aqualite.jp/
* AquaLight
* AquaFizz
* AquaFine
* AquaCube
* AquaSlim
* BaseAqua
* TinyAqua

**Comment by suzuki-shunsuke** [comment]
> I would like to keep names of GitHub Organization, Repository, and command.
> So how about combining any word and Aqua?

I changed my mind.
I don't stick to the original name "aqua".

`clisvm` looks unique.

https://twitter.com/search?q=clisvm&src=typed_query&f=live
https://www.google.co.jp/search?q=clisvm

**Comment by suzuki-shunsuke** [comment]
`clienv`
https://twitter.com/search?q=clienv&src=spelling_expansion_revert_click&f=live
https://github.com/search?q=clienv
https://www.google.co.jp/search?q=clienv

**Comment by suzuki-shunsuke** [comment]
`clienv` may be confusing with `client`.

**Comment by suzuki-shunsuke** [comment]
https://docs.github.com/en/organizations/managing-organization-settings/renaming-an-organization

**Comment by suzuki-shunsuke** [comment]
## Open PRs

- [ ] aqua
- [ ] aqua-registry
- [ ] aqua-installer
- [ ] aqua-renovate-config
- [ ] aquaproj.github.io
- [ ] circleci-orb-aqua
- [ ] homebrew-aqua
- [ ] aqua-proxy
- [ ] test-aqua-renovate-config
- [ ] demo
- [ ] .github

## Publish the migration guide

- [ ] GitHub Issue

## Rename Organizations

- [ ] Rename Organizations

## Rename repositories

- [ ] aqua
- [ ] aqua-registry
- [ ] aqua-installer
- [ ] aqua-renovate-config
- [ ] aquaproj.github.io
- [ ] circleci-orb-aqua
- [ ] homebrew-aqua
- [ ] aqua-proxy
- [ ] test-aqua-renovate-config

## Create releases

- [ ] aqua-proxy
- [ ] aqua
- [ ] aqua-registry
- [ ] aqua-installer
- [ ] aqua-renovate-config
- [ ] circleci-orb-aqua

**Comment by suzuki-shunsuke** [comment]
```sh
#!/usr/bin/env bash

set -eux
set -o pipefail

git_replace() {
  git grep -l "$1" | xargs -n 1 gsed -i "s|$1|$2|g"
}

git_replace aquaproj-aqua clivm
git_replace aquaproj clivm
git_replace AQUA_ CLIVM_
git_replace aqua clivm
```

**Comment by suzuki-shunsuke** [comment]
https://github.com/orgs/aquaproj/projects/6?add_cards_query=is%3Aopen

**Comment by suzuki-shunsuke** [comment]
https://docs.brew.sh/Rename-A-Formula

**Comment by suzuki-shunsuke** [comment]
I'm planning to rename `aqua` to `clivm`. What do you think?
https://twitter.com/search?q=clivm&src=typed_query&f=live
https://github.com/search?q=clivm
https://www.google.co.jp/search?q=clivm

**Comment by suzuki-shunsuke** [comment]
# Migration Guide

## Laptop

* Install clivm
* Update environment variables: https://aquaproj.github.io/docs/reference/config#environment-variables
  * `PATH`
  * `AQUA_*`
* (Optional) Copy the root directory `AQUA_ROOT_DIR`
* Copy `aqua.yaml` to `clivm.yaml`
* Open a new terminal and confirm clivm works well
* Remove the old root directory
* Remove `aqua.yaml`

## Repository

* Replace words
  * `aquaproj-aqua`: `clivm`
  * `aquaproj`: `clivm`
  * `AQUA`: `CLIVM`
  * `aqua`: `clivm`
* Rename files
  * `aqua.yaml`: `clivm.yaml`

**Comment by suzuki-shunsuke** [comment]
# [DRAFT] Announcement: `aqua` will be renamed to `clivm`

⚠️ The announcement is still **draft**.

The project `aqua` will be renamed to `clivm`, because the word `aqua` is so common that the googleability of aqua is low.
It is difficult to search the information about aqua using Google or another similar Internet search engine.

By renaming `aqua` to more unique name, the googleability of this project will be improved.

We will change the following things.

* Rename the project name `aqua` to `clivm`
* Rename GitHub Organizations `aquaproj` to `clivm`
* Rename GitHub Repositories under aquaproj: e.g. Rename aqua-installer to clivm-installer
* Change the official website URL https://aquaproj.github.io to https://clivm.github.io
* Change the CLI names
  * `aqua` => `clivm`
  * `aqua-installer` => `clivm-installer`
  * `aqua-proxy` => `clivm-proxy`
* Change the default root directory `AQUA_ROOT_DIR`
  * linux and macOS: `${XDG_DATA_HOME:-$HOME/.local/share}/aquaproj-aqua` => `${XDG_DATA_HOME:-$HOME/.local/share}/clivm`
  * windows: `${HOME/AppData/Local}/aquaproj-aqua` => `${HOME/AppData/Local}/clivm`
* Change the prefix of environment variable names from `AQUA_` to `CLIVM_`
  * e.g. `AQUA_ROOT_DIR` => `CLIVM_ROOT_DIR`
* Rename configuration files: `\.?aqua\.ya?ml` => `\.?clivm\.ya?ml`

## What should you do?

### Laptop

* Install clivm: https://aquaproj.github.io/docs/reference/install
* Update environment variables: https://aquaproj.github.io/docs/reference/config#environment-variables
  * `PATH`
  * `AQUA_*`
* (Optional) Copy the root directory `AQUA_ROOT_DIR`
* Copy `aqua.yaml` to `clivm.yaml`
* Open a new terminal and confirm clivm works well
* Remove the old root directory
* Remove `aqua.yaml`

### Repository

Basically, you can fix codes with problem.

* Replace words
  * `aquaproj-aqua`: `clivm`
  * `aquaproj`: `clivm`
  * `AQUA`: `CLIVM`
  * `aqua`: `clivm`
* Rename files
  * `aqua.yaml`: `clivm.yaml`

**Comment by 4513ECHO** [comment]
I'm sorry for late, but I'm going to write my opinion for this proposal as a mere user.
We should rename it. It's not good that the googlability is low. And the googlability of `aqua` is actually low.

On the other hand, I like the atmosphere of the name of `aqua`.
I also like the aqua logo. One of the reason why I started using aqua is not only the declarative convenience but also the name.

The coined word has higher googlability than abbreviation. I suggest `aqucliva`. It's a word that mixes `aqua` and `cli`. It's unique, and I think the atmosphere of the original name is maintained.

https://twitter.com/search?q=aqucliva&src=typed_query&f=live
 https://www.google.co.jp/search?q=aqucliva
 https://github.com/search?q=aqucliva&type=repositories

What do you think?

**Comment by suzuki-shunsuke** [comment]
Thank you for your opinion.

> We should rename it. It's not good that the googlability is low. And the googlability of aqua is actually low.
> 
> On the other hand, I like the atmosphere of the name of aqua.
> I also like the aqua logo. One of the reason why I started using aqua is not only the declarative convenience but also the name.
> 
> clivm is a name that only represents features and is too tasteless.
> 
> The coined word has higher googlability than abbreviation.

I agree you.
I also like the atmosphere of the name of aqua and logo.

`aqucliva` looks cool, but I have two concerns about `aqucliva`.

1. It is a little long. Long name is not desirable for typing. But maybe this is not so import
2. I can't guess the pronunciation of `aqucliva`

**Comment by suzuki-shunsuke** [comment]
I reconsidered new names related to `aqua`, but I think `clivm` is better than them.
`clivm` may be tasteless, but it is short, unique, and easy to remember.

I really appreciate your opinion, and I'm sorry that I can't adopt your proposal.
I'm glad that you like aqua's name and logo.
I also like them.

Thank you for your proposal.

**Comment by tadayosi** [comment]
To me renaming to either of `aquacli`, `aquaclivm`, and `aquavm` seems to be enough. And I'd prefer keeping the rest of config files and env var names as-is.

**Comment by suzuki-shunsuke** [comment]
Thank you for your opinion.

`aquaclivm` is a little too long.
`aquavm` could be nice.

It is hard to make a decision 😥

## aquacli

 https://www.google.co.jp/search?q=aquacli
https://twitter.com/search?q=aquacli
 https://github.com/search?q=aquacli

## aquaclivm

 https://www.google.co.jp/search?q=aquaclivm
https://twitter.com/search?q=aquaclivm
 https://github.com/search?q=aquaclivm

## aquavm

 https://www.google.co.jp/search?q=aquavm
https://twitter.com/search?q=aquavm
 https://github.com/search?q=aquavm

⚠️ There is already a project `aquavm`. https://github.com/fluencelabs/aquavm

**Comment by suzuki-shunsuke** [comment]
If we rename aqua to `aquavm`, we may only have to update document, and could keep the command name, GitHub Organization and Repository names, the official website URL, environment variables, configuration file names, and so on.

**Comment by suzuki-shunsuke** [comment]
👍 `aquavm` 2,080
👎 `Aqua VM` 29,100,000

<img width="844" alt="image" src="https://user-images.githubusercontent.com/13323303/176433799-e155df5a-fa0a-488f-8dac-b231af7eb14f.png">

--

<img width="931" alt="image" src="https://user-images.githubusercontent.com/13323303/176433974-6c3fee69-36cc-4756-bd28-f75833944178.png">

**Comment by takumin** [comment]
How about `aquamgr`?

 https://www.google.co.jp/search?q=aquamgr
https://twitter.com/search?q=aquamgr
 https://github.com/search?q=aquamgr

Thank you very much for this great tool!

**Comment by budimanjojo** [comment]
What if you rename the GH organization to `aquavm/aqua`? Does that work?

**Comment by JJGadgets** [comment]
How about `aquapkg`? Gets the point across that Aqua deals with packages, and not as awkward to say to other people/out loud.

**Comment by Gowiem** [comment]
Just stumbled upon this. I like the name aqua and don't know if googleability is the biggest issue I would be concerned with. Getting the word out more seems like it would be more important than a big rename migration? Just my humble opinion.

**Comment by suzuki-shunsuke** [comment]
Dear folks,

Thank you for your opinion.
I close this issue now because I decided not to rename the project now.
I apologize for wonderful names you have come up with.

aqua is now used by more users and projects than ever before, so I don't want to confuse users by renaming this project.
I should have rename this project more earlier, but I couldn't make a decision and lost a chance.

I'd like to consider how to raise awareness of this project instead.

- https://github.com/aquaproj/aqua/issues?q=is%3Aissue+is%3Aopen+awareness+label%3Araise-awareness
- https://github.com/orgs/aquaproj/discussions/1650
